### PR TITLE
Re-generate Compute cmdlets with PS object output type support

### PR DESCRIPTION
**Pull request by hyonholee**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: tools/StaticAnalysis/Exceptions/BreakingChangeIssues.csv [hunk 1]
```diff
@@ -121,4 +121,43 @@
...
-"D:\workspace\powershell-demand\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Compute\Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.Automation.ConvertToAzureRmVMManagedDisk","ConvertTo-AzureRmVMManagedDisk","0","1050","The parameter set 'InvokeByDynamicParameters' for cmdlet 'ConvertTo-AzureRmVMManagedDisk' has been removed.","Add parameter set 'InvokeByDynamicParameters' back to cmdlet 'ConvertTo-AzureRmVMManagedDisk'."
\ No newline at end of file
+"D:\workspace\powershell-demand\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Compute\Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.Automation.ConvertToAzureRmVMManagedDisk","ConvertTo-AzureRmVMManagedDisk","0","1050","The parameter set 'InvokeByDynamicParameters' for cmdlet 'ConvertTo-AzureRmVMManagedDisk' has been removed.","Add parameter set 'InvokeByDynamicParameters' back to cmdlet 'ConvertTo-AzureRmVMManagedDisk'."
```
Comment (huangpf) [inline tools/StaticAnalysis/Exceptions/BreakingChangeIssues.csv]: Why are these added in your PR?

File: src/ResourceManager/Compute/Commands.Compute/Manual/SetAzureRmSnapshotUpdateImageReferenceCommand.cs [hunk 2]
```diff
@@ -14,18 +14,10 @@
 // limitations under the License.
 // 
 
-// Warning: This code was generated by a tool.
```
Comment (huangpf) [inline src/ResourceManager/Compute/Commands.Compute/Manual/SetAzureRmSnapshotUpdateImageReferenceCommand.cs]: This command is not auto-generated anymore?

File: src/ResourceManager/Compute/Commands.Compute/Manual/SetAzureRmDiskUpdateImageReferenceCommand.cs [hunk 3]
```diff
@@ -14,18 +14,10 @@
 // limitations under the License.
 // 
 
-// Warning: This code was generated by a tool.
```
Comment (huangpf) [inline src/ResourceManager/Compute/Commands.Compute/Manual/SetAzureRmDiskUpdateImageReferenceCommand.cs]: What happened to these commands?

Comment (hyonholee) [inline src/ResourceManager/Compute/Commands.Compute/Manual/SetAzureRmDiskUpdateImageReferenceCommand.cs]: This is under \Manual folder.  So I removed it.

File: src/ResourceManager/Compute/Commands.Compute/Manual/SetAzureRmSnapshotUpdateImageReferenceCommand.cs (hunk 2 above)
Comment (hyonholee) [inline src/ResourceManager/Compute/Commands.Compute/Manual/SetAzureRmSnapshotUpdateImageReferenceCommand.cs]: This is under \Manual folder.

File: tools/StaticAnalysis/Exceptions/BreakingChangeIssues.csv (hunk 1 above)
Comment (hyonholee) [inline tools/StaticAnalysis/Exceptions/BreakingChangeIssues.csv]: This is not added.  This line was at the end of the file without new line before, so it appears this line is modified.

**Comment by huangpf** [comment]
LGTM.

**Comment by markcowl** [comment]
@hyonholee there are merge conflicts.  Also, does this include the cmdlet changes from the hotfix?

**Comment by hyonholee** [comment]
@markcowl I resolved merge conflicts, and added the changes from hotfix.  Also I included Evgeny's alias update.

File: src/ResourceManager/Compute/Commands.Compute/help/New-AzureRmVMSqlServerKeyVaultCredentialConfig.md [hunk 4]
```diff
@@ -96,44 +96,12 @@ Parameter Sets: (All)
...
 
 Required: False
-Position: Named
+Position: 0
 Default value: None
 Accept pipeline input: True (ByPropertyName)
 Accept wildcard characters: False
 ```
 
-### -Confirm
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute/help/New-AzureRmVMSqlServerKeyVaultCredentialConfig.md]: Why are these removed?

File: src/ResourceManager/Compute/Commands.Compute/help/New-AzureRmVMSqlServerAutoBackupConfig.md [hunk 5]
```diff
@@ -115,7 +115,7 @@ Parameter Sets: (All)
 Aliases: 
 
 Required: False
-Position: 4
+Position: 3
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute/help/New-AzureRmVMSqlServerAutoBackupConfig.md]: why are the positions changed?

**Comment by hyonholee** [comment]
@markcowl I resolved the conflict and fixed the wrong update of help documents.

**Comment by cormacpayne** [comment]
On-demand: http://azuresdkci.cloudapp.net/job/powershell-demand/1761/

File: src/ResourceManager/Compute/Commands.Compute/Generated/AvailabilitySet/AvailabilitySetGetMethod.cs [hunk 6]
```diff
@@ -83,20 +82,13 @@ protected void ExecuteAvailabilitySetGetMethod(object[] invokeMethodInputParamet
...
             {
                 var result = AvailabilitySetsClient.List(resourceGroupName);
                 var resultList = result.ToList();
-                var psObject = new List<PSAvailabilitySetList>();
-                foreach (var r in resultList)
-                {
-                    psObject.Add(Mapper.Map<AvailabilitySet, PSAvailabilitySetList>(r));
-                }
-                WriteObject(psObject, true);
+                WriteObject(resultList, true);
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Generated/AvailabilitySet/AvailabilitySetGetMethod.cs]: @hyonholee it looks like this cmdlet going to return `AvailabilitySet` instead of `PSAvailabilitySet` now. What was the reason for this?

File: src/ResourceManager/Compute/Commands.Compute/Generated/Disk/DiskCreateOrUpdateMethod.cs [hunk 7]
```diff
@@ -62,7 +62,7 @@ protected object CreateDiskCreateOrUpdateDynamicParameters()
 
             var pDisk = new RuntimeDefinedParameter();
             pDisk.Name = "Disk";
-            pDisk.ParameterType = typeof(PSDisk);
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Generated/Disk/DiskCreateOrUpdateMethod.cs]: @hyonholee do the changes made to this file revert the breaking change mitigation from last release? (_i.e._, using `PSDisk` instead of `Disk`)

File: src/ResourceManager/Compute/Commands.Compute/Generated/AvailabilitySet/AvailabilitySetGetMethod.cs (hunk 6 above)
Comment (hyonholee) [inline src/ResourceManager/Compute/Commands.Compute/Generated/AvailabilitySet/AvailabilitySetGetMethod.cs]: This code does not produce Powershell cmdlets.  The method is for the generalized Powershell cmdlets (Invoke cmdlets), but we don't generate this cmdlets anymore.  So this change does not have any effect.

File: src/ResourceManager/Compute/Commands.Compute/Generated/Disk/DiskCreateOrUpdateMethod.cs (hunk 7 above)
Comment (hyonholee) [inline src/ResourceManager/Compute/Commands.Compute/Generated/Disk/DiskCreateOrUpdateMethod.cs]: No.  The real change is in ProcessRecord() method.  The method, CreateDiskCreateOrUpdateDynamicParameters() and ExecuteDiskCreateOrUpdateMethod() are for the generalized cmdlet (Invoke cmdlets), but we no longer generated the cmdlets.   So anything in these methods do not have any effect on cmdlets.

**Comment by hyonholee** [comment]
@cormacpayne @markcowl   Could you merge this PR if there is no other issue?  I need to send another PR, but I want this one merged before that.  Thanks.

**Comment by cormacpayne** [comment]
On-demand: http://azuresdkci.cloudapp.net/job/powershell-demand/1769/
